### DISCUSSION TITLE
Le pattern Memento

### DISCUSSION THREAD

**Discussion by giscus** [zero-filtre/zerofiltre-blog#169] [Announcements]
# Le pattern Memento

Nous allons voir dans quel contexte on utilise le pattern Memento et comment l'implémenter en Java.

https://zerofiltre.tech/articles/259

**Comment by imphilippesimo** [comment]
# Commentaires d'avant publication 

## Image principale: 
Je te recommande de la faire sous [canva](https://www.canva.com/en_gb/)

## a/ Classification des design patterns
Il serait bien d'ajouter des exemples de patterns de chaque catégorie.

Exemple: 
```` 
Les exemples courants de Design Patterns comportementaux incluent le Modèle Observateur, 
le Modèle Stratégie et le Modèle Commande.
```` 

## b/ Pourquoi apprendre les design patterns?

*Il vous enseigne des techniques qui permettent de résoudre plusieurs types de problèmes en utilisant les principes de la programmation orientée objet....*  

J'aurais dit : 


*Il vous enseigne des techniques qui permettent de résoudre plusieurs types de problèmes en utilisant les principes de séparation des préoccupations, d'encapsulation, de modularité et de réutilisabilité du code,*    

Car en effet les Design Patterns ne s'appliquent pas qu'à la POO

**Reply by imphilippesimo** [reply]
J'ai failli oublier: 
Tout code source présent dans un article doit être fusionné au projet Github articles.
[Je te laisse prendre connaissance de la procédure](https://github.com/Zerofiltre-Courses/articles/blob/main/README.md)
